test/e2e/infinispan/upgrade_operand_test.go

```diff
@@ -422,6 +424,35 @@ func TestPodAlreadyShutdownOnUpgrade(t *testing.T) {
 	assert.EqualValues(t, int32(0), ss.Status.UpdatedReplicas)
 }
 
+func TestScaleDownBlockedWithDegradedCache(t *testing.T) {
+	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
+
+	replicas := 1
+	ispn := tutils.DefaultSpec(t, testKube, func(i *ispnv1.Infinispan) {
+		i.Spec.Replicas = int32(replicas)
+	})
+	testKube.CreateInfinispan(ispn, tutils.Namespace)
+	testKube.WaitForInfinispanPods(replicas, tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	ispn = testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionWellFormed)
+
+	_client := tutils.HTTPClientForClusterWithVersionManager(ispn, testKube, tutils.VersionManager())
+	cacheName := "cache"
+	cacheConfig := "<distributed-cache><partition-handling when-split=\"DENY_READ_WRITES\" merge-policy=\"PREFERRED_ALWAYS\"/></distributed-cache>"
+	cacheHelper := tutils.NewCacheHelper(cacheName, _client)
+	cacheHelper.Create(cacheConfig, mime.ApplicationXml)
+	cacheHelper.Available(false)
+
+	tutils.ExpectNoError(
+		testKube.UpdateInfinispan(ispn, func() {
+			ispn.Spec.Replicas = 0
+		}),
+	)
+	testKube.WaitForInfinispanState(ispn.Name, ispn.Namespace, func(i *ispnv1.Infinispan) bool {
```

Comment:
Should we extend the test for the second part, where we fix the degraded cache and verify the upgrade will proceed further?

Reply:
I have updated the test so that we set the Cache to AVAILABLE and then we wait for the cluster to scale down to 0 pods with `GracefulShutdown=True`. We haven't changed the semantics of the rest of the GracefulShutdown upgrade procedure, so it's not necessary to change the remainder of the upgrade workflow.